src/menu/Menu.ts

```diff
@@ -0,0 +1,311 @@
+import { assign } from '@dojo/core/lang';
+import { createTimer } from '@dojo/core/util';
+import uuid from '@dojo/core/uuid';
+import { Handle } from '@dojo/interfaces/core';
+import { v, w } from '@dojo/widget-core/d';
+import { DNode } from '@dojo/widget-core/interfaces';
+import ThemeableMixin, { theme, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import MenuItem, { MenuItemProperties } from './MenuItem';
+import * as css from './styles/menu.css';
+
+export type Role = 'menu' | 'menubar';
+
+export interface MenuProperties extends ThemeableProperties {
+	/**
+	 * Only applies to nested menus. Either a flag indicating whether the widget instance should handle animating
+	 * between the visible and hidden states, or a function that manually handles the animation. If true (the default),
+	 * then the menu will slide into and out of view like an accordion. If false, then any animation should be handled
+	 * in the CSS, as the menu will just snap open/shut.
+	 */
+	animate?: boolean | ((element: HTMLElement, hidden: boolean) => void);
+
+	/**
+	 * Indicates whether the menu is disabled. If true, then the widget will ignore events.
+	 */
+	disabled?: boolean;
```

Comment:
If `label` is specified, then a `MenuItem` is created and `disabled` is passed to it, which keeps it from being expanded.